### PR TITLE
Feature/add data model

### PR DESCRIPTION
**Pull request by AnthonySchuijlenburg**
This pr adds the data models

To test it run:
`php artisan migrate:fresh --seed`

### CODE REVIEW

**Comment by JaZo** [comment]
:information_source: Now with type hints in your code, defining types within PHP DocBlocks is redundant. Laravel has removed all of the `@param` and `@return` tags from its DocBlocks where types are defined with PHP. Similarly, Shift [removed these tags](https://github.com/swisnl/dlf-hackathon-2023/pull/10/commits/8c9f2ece8c9419454aeb211e653cace2cf2a78a7?w=1) from any DocBlock where the code now has _equivalent_ type hints.

**Comment by JaZo** [comment]
:information_source: While the _Laravel Fixer_ performs automation to adopt the latest Laravel conventions, Shift understands developers have different preferences. All of Shift's automation is done in nice, atomic commits. This makes it easier to undo any of the changes Shift makes using `git revert`.

File: app/Models/Tenant.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public function sluggable(): array
+    {
+        return [
+            'slug' => [
+                'source' => 'name',
+            ],
+        ];
+    }
+
+    public function transaction(): HasMany
```
Comment (JaZo) [inline app/Models/Tenant.php]: This must be plural.

File: database/factories/AccountFactory.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        return [
+            'email' => Hash::make($this->faker->unique()->safeEmail()),
+        ];
+    }
+
+    public function hasTransaction(): static
```
Comment (Rocksheep) [inline database/factories/AccountFactory.php]: THis is not needed as Laravel magic already does this for you

File: database/factories/TenantFactory.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+     */
+    public function definition(): array
+    {
+        return [
+            'name' => $this->faker->company(),
+            'domain' => $this->faker->url(),
+        ];
+    }
+
+    public function hasTransaction(): static
```
Comment (Rocksheep) [inline database/factories/TenantFactory.php]: Idem

File: app/Models/Transactions.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+
+class Transactions extends Model
+{
+    use HasFactory;
+    use HasUuids;
+
+    protected $fillable = [
+        'grams_of_co2',
```
Comment (JaZo) [inline app/Models/Transactions.php]: I think we should add the amount (EUR) we charged.

File: app/Models/Account.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+class Account extends Model
+{
+    use HasFactory;
+    use HasUuids;
+
+    protected $fillable = [
+        'email',
+    ];
+
+    public function transaction(): HasMany
```
Comment (JaZo) [inline app/Models/Account.php]: This must be plural.

**Review by JaZo — CHANGES_REQUESTED** [review]

**Review by Rocksheep — CHANGES_REQUESTED** [review]

File: composer.json [hunk 6]
```diff
@@ -49,8 +50,11 @@
         "analyse": [
             "vendor/bin/phpstan analyse --ansi"
         ],
-        "format": [
-            "vendor/bin/pint"
+        "check-style": [
+            "./vendor/bin/pint --test"
+        ],
+        "fix-style": [
+            "./vendor/bin/pint "
```
Comment (JaZo) [inline composer.json]: There's an extraneous space at the end.

**Review by JaZo — CHANGES_REQUESTED** [review]

File: app/Models/Transactions.php
Comment (JaZo) [inline app/Models/Transactions.php:1]: The model name should be singular.

**Review by JaZo — CHANGES_REQUESTED** [review]

File: app/Models/Account.php [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    use HasFactory;
+    use HasUuids;
+
+    protected $fillable = [
+        'email',
+    ];
+
+    public function transactions(): HasMany
+    {
+        return $this->hasMany(Transactions::class);
```
Comment (Rocksheep) [inline app/Models/Account.php]: This transactions class is singular

File: app/Models/Tenant.php [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        return [
+            'slug' => [
+                'source' => 'name',
+            ],
+        ];
+    }
+
+    public function transactions(): HasMany
+    {
+        return $this->hasMany(Transactions::class);
```
Comment (Rocksheep) [inline app/Models/Tenant.php]: Same here

File: app/Models/Transactions.php
Comment (Rocksheep) [reply app/Models/Transactions.php:1]: I agree

**Review by Rocksheep — CHANGES_REQUESTED** [review]

**Review by Rocksheep — APPROVED** [review]

**Pull request merged**